numpy/_core/shape_base.py

```diff
@@ -306,9 +308,11 @@ def hstack(tup, *, dtype=None, casting="same_kind"):
 
     Parameters
     ----------
-    tup : sequence of ndarrays
+    tup : ndarray or sequence of ndarrays
         The arrays must have the same shape along all but the second axis,
-        except 1-D arrays which can be any length.
+        except 1-D arrays which can be any length. In the case of a single
+        array_like input, it will be treated as a sequence of `shape[0]`
+        arrays, each of shape `shape[1:]`.
```

Comment:
```suggestion
        array_like input, it will be treated as a sequence of arrays; i.e.,
        each element along the zeroth axis is treated as a separate array.
```